openpype/hosts/maya/plugins/publish/extract_look.py

```diff
@@ -133,6 +89,307 @@ def no_workspace_dir():
+
+    @staticmethod
+    def get_extension():
+        pass
+
+    def __repr__(self):
+        # Log instance as class name
+        return self.__class__.__name__
+
+
+class MakeRSTexBin(TextureProcessor):
+    """Make `.rstexbin` using `redshiftTextureProcessor`"""
+
+    def process(self,
+                source,
+                colorspace,
+                color_management,
+                staging_dir):
+
+        texture_processor_path = self.get_redshift_tool(
+            "redshiftTextureProcessor"
+        )
+        if not texture_processor_path:
+            raise KnownPublishError("Must have Redshift available.")
+
+        subprocess_args = [
+            texture_processor_path,
+            source
+        ]
+
+        hash_args = ["rstex"]
+        texture_hash = source_hash(source, *hash_args)
+
+        # Redshift stores the output texture next to the input but with
+        # the extension replaced to `.rstexbin`
+        basename, ext = os.path.splitext(source)
+        destination = "{}{}".format(basename, self.get_extension())
+
+        self.log.debug(" ".join(subprocess_args))
+        try:
+            run_subprocess(subprocess_args)
+        except Exception:
+            self.log.error("Texture .rstexbin conversion failed",
+                           exc_info=True)
+            raise
+
+        return TextureResult(
+            path=destination,
+            file_hash=texture_hash,
+            colorspace=colorspace,
+            transfer_mode=COPY
+        )
+
+    @staticmethod
+    def get_extension():
+        return ".rstexbin"
+
+    @staticmethod
+    def get_redshift_tool(tool_name):
+        """Path to redshift texture processor.
+
+        On Windows it adds .exe extension if missing from tool argument.
+
+        Args:
+            tool_name (string): Tool name.
+
+        Returns:
+            str: Full path to redshift texture processor executable.
+        """
+        if "REDSHIFT_COREDATAPATH" not in os.environ:
+            raise RuntimeError("Must have Redshift available.")
+
+        redshift_tool_path = os.path.join(
+            os.environ["REDSHIFT_COREDATAPATH"],
+            "bin",
+            tool_name
+        )
+
+        return find_executable(redshift_tool_path)
+
+
+class MakeTX(TextureProcessor):
+    """Make `.tx` using `maketx` with some default settings.
+
+    Some hardcoded arguments passed to `maketx` are based on the defaults used
+    in Arnold's txManager tool.
+
+    """
+
+    def __init__(self, log=None):
+        super(MakeTX, self).__init__(log=log)
+        self.extra_args = []
+
+    def apply_settings(self, system_settings, project_settings):
+        # Allow extra maketx arguments from project settings
+        extra_args_dict = (
+            project_settings["maya"]["publish"]
+            .get("ExtractLook", {}).get("maketx_arguments", {})
+        )
+        extra_args = []
+        for flag, value in extra_args_dict.items():
+            extra_args.append(flag)
```

Comment:
tiny but:
```suggestion
            extra_args.extend([flag, value])
```

Reply:
I've actually refactored this to support flags without values too, which `--opaque-detect` technically is. I'm wondering whether the `dict` structure in settings is actually best and whether it might make more sense to just make it a list for strings so that if values would be needed one could just add `--flag-with-value`, `1024` as separate keys. Then that might also make it work if there are ever flags for `maketx` that take more than one value which e.g. `--colorconvert` already is.

Thoughts?

Reply:
Or simplify it to just argument and list of parameters
```json
{
    "type": "list",
    "key": "maketx_args",
    "object_type": {
        "type": "dict",
        "children": [
            {
                "key": "argument",
                "label": "Argument",
                "type": "text"
            },
            {
                "key": "list",
                "label": "Parameters",
                "object_type": "text"
            }
        ]
    }
}
```

This allows 0-n parameters for an argument

Reply:
I like it, thanks - will play with it.

Reply:
So this doesn't work:
```
{
                    "key": "maketx_arguments",
                    "label": "MakeTX Arguments",
                    "collapsible": true,
                    "type": "list",
                    "object_type": {
                        "type": "dict",
                        "children": [
                            {
                                "key": "argument",
                                "label": "Argument",
                                "type": "text"
                            },
                            {
                                "key": "list",
                                "label": "Parameters",
                                "type": "text"
                            }
                        ]
                    }
                }
```
The parameters don't turn into a list and the other example I couldn't get to work.

Reply:
+1 on the list of strings

Reply:
I don't like list of strings because 99% user have not idea how to split them into separated arguments.

> The parameters don't turn into a list

I made mistake it should be 
```
{
    "type": "list",
    "key": "maketx_args",
    "object_type": {
        "type": "dict",
        "children": [
            {
                "key": "argument",
                "label": "Argument",
                "type": "text"
            },
            {
                "type": "list",
                "label": "Parameters",
                "object_type": "text"
            }
        ]
    }
}
```

Reply:
is that any different than the existing ffmpeg args? I'd argue that that 99% of the users that don't know how to split CLI arguments won't/shouldn't be touching any of the arguments

Reply:
Current ffmpeg arguments must be converted to string and the process is started using `shell=True` because of that -> that cause more issues than you would think.

Reply:
This `'"/path/to/custom/font/with/space myfont.ttf"'` is actually invalid argument because it does not have spaces or special characters it will be passed as string `"/path/to/custom/font/with/space myfont.ttf"`, meaning the quotes will be part of the value passed to process.

Reply:
EDITED: It has space, but in that case you must know when the quotes should be there or not.

Reply:
At the end, I don't care, I'm just trying to explain why I think it is better to do it as I proposed.

With the proposed settings you can easily explanain how the arguments should be passed and how it works, instead of trying to figure out how to guess what user meant by passing values there.

Not sure if it's clear from previous messages but shlex and processing of subprocess arguments does not work the same way for all platforms.

Reply:
For now I've decided to go for your proposal @iLLiCiTiT thanks. Commits coming up!

Reply:
That looks like this:
![afbeelding](https://user-images.githubusercontent.com/2439881/228507210-4abf5c51-06ea-4974-b201-78e83458f20b.png)